tools/enrollment_preprocessor.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+import csv
+import sys
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from io import BytesIO
+from typing import TextIO
+
+from openpyxl import load_workbook
+from openpyxl.cell import Cell
+
+
+@dataclass
+class User:
+    title: str
+    last_name: str
+    first_name: str
+    email: str
+
+
+@dataclass
+class UserCells:
+    title: Cell | None
+    last_name: Cell
+    first_name: Cell
+    email: Cell
+
+    def value(self) -> User:
+        return User(
+            str(self.title.value) or "" if self.title else "",
+            str(self.last_name.value),
+            str(self.first_name.value),
+            str(self.email.value),
+        )
+
+
+def user_decision(field: str, existing: str, imported: str) -> str:
+    if existing == imported:
+        return existing
+    decision = input(f"Do you want to keep the existing user {field}? (y/n) ")
+    if decision and decision[0].lower() == "n":
+        return imported
+    return existing
+
+
+def fix_user(users: dict[str, User], imported_cells: UserCells) -> None:
+    imported = imported_cells.value()
+    existing = users.setdefault(imported.email, imported)
+    if imported == existing:
+        return
+    print("There is a conflict in the user data.")
+    print(f"existing: {existing}.")
+    print(f"imported: {imported}.")
+    if imported_cells.title is not None:
+        imported_cells.title.value = user_decision("title", existing.title, imported.title)
+    imported_cells.last_name.value = user_decision("last name", existing.last_name, imported.last_name)
+    imported_cells.first_name.value = user_decision("first name", existing.first_name, imported.first_name)
+    imported_cells.email.value = user_decision("email", existing.email, imported.email)
+    print()
+
+
+def run_preprocessor(enrollment_data: str | BytesIO, user_data: TextIO) -> None:
```

Comment:
Can `enrollment_data` really be a `str`? I'd expect `load_workbook` to call `read` on it. Maybe [`typing.IO`](https://docs.python.org/3/library/typing.html#typing.IO) is the right annotation here (since I'm in favor of not modifying the argument but returning a new `BytesIO` instance).

Reply:
yes, it is the PATH to the excel file. Why? Idk. Otherwise, I would need to wrap TextIO/BinaryIO to BytesIO for the cli